Helpers/WithContext.php

```diff
@@ -0,0 +1,40 @@
+<?php
+namespace Exceptions\Helpers;
+
+/**
+ * Implements a set of helper methods that creates exceptions to attach and retrieve contextual data.
+ *
+ * @package Exceptions\Helpers
+ */
+trait WithContext
+{
+    protected $context = null;
```

Comment:
Missing phpdoc?

```php
/** @var mixed */
protected $context = null;
```

Reply:
If thats all, i'll merge right after

Reply:
That's all I see; other changes you made seems to be OK for me :)

Reply:
Well... I just saw something might be missing: the previous exception =/